vuestorefront/models/ir_ui_view.py

```diff
@@ -0,0 +1,42 @@
+from odoo import models
+
+from ..utils import format_vsf_cache_tags, invalidate_vsf_cache
+from .. import const
+
+
+# Should we use lower case? Because sending this to VSF cache using
+# params will make it lowercase anyway..
+TAG_PFX = "IUV"
+
+
+class IrUiView(models.Model):
+    _inherit = 'ir.ui.view'
+
+    def get_vsf_cache_tags(self):
+        return format_vsf_cache_tags(TAG_PFX, self.ids)
+
+    def write(self, vals):
+        res = super().write(vals)
+        self.invalidate_website_page_views_cache()
+        return res
+
+    def unlink(self):
+        self.invalidate_website_page_views_cache()
+        return super().unlink()
```

Comment:
Wondering if would not be from performance perspective good if this methods are not even called if there is no page related to the view?

Also is the `website.page` `write` called when the views for the page are updated, if it is maybe would be better to do the invalidation from that model?

Reply:
It only invalidates views that are related with pages. So if view is not related, it will be ignored, even if it was called.

Regarding second question, we can't check that via `website.page`, because it does not have any content with actual view. Its only related with it. But content generation happens from view. So if website page view changes, we then do cache invalidation. And we must check CRUD on that model, not on page (because we are caching view (content), not the page itself).